src/flow/persistence/mod.rs

```diff
@@ -253,6 +286,64 @@ impl GroupCommitQueueSet {
             .min()
     }
 
+    /// Retrieves a vector of packets from the persistent log.
+    pub fn retrieve_recovery_packets(&self, nodes: &DomainNodes) -> Vec<Box<Packet>> {
+        let mut packets = vec![];
+        for (_index, node) in nodes.iter() {
+            let node = node.borrow();
+            let local_addr = node.local_addr();
+            let global_addr = node.global_addr();
+            let path = self.log_path(&local_addr);
+            if !path.exists() {
+                continue;
+            }
+
+            let file = File::open(path).unwrap();
```

Comment:
Instead of checking whether the file exists and then unwrapping, it may be better to just try opening the log and catching any errors that triggers.